SDL_gpu_shadercross.h

```diff
@@ -38,23 +38,430 @@ SDL_GpuShaderFormat SDL_ShaderCross_GetShaderFormats(void)
+    );
+
+    HRESULT(__stdcall *Disassemble)(
+        IDxcCompiler3 *This,
+        const DxcBuffer *pObject,
+        REFIID riid,
+        LPVOID *ppResult
+    );
+} IDxcCompiler3Vtbl;
+struct IDxcCompiler3
+{
+    const IDxcCompiler3Vtbl *lpVtbl;
+};
+
+/* DXCompiler */
+static void *dxcompiler_dll = NULL;
+
+typedef HRESULT (__stdcall *DxcCreateInstanceProc)(
+    REFCLSID rclsid,
+    REFIID riid,
+    LPVOID *ppv);
+
+static DxcCreateInstanceProc SDL_DxcCreateInstance = NULL;
+static IDxcCompiler3 *SDL_DxcInstance = NULL;
+
+static void *SDL_ShaderCross_INTERNAL_CompileDXC(
+    SDL_GpuDevice *device,
+    void *createInfo,
+    const char *hlslSource,
+    const char *shaderProfile,
+    SDL_bool spirv)
+{
+    DxcBuffer source;
+    IDxcResult *dxcResult;
+    DXC_OUT_KIND blobOut;
+    IDxcBlob *blob;
+    IDxcBlobUtf8 *errors;
+    LPCWSTR args[] = {
+        L"-E",
+        L"main", /* FIXME */
+        NULL,
+        NULL,
+        NULL
+    };
+    Uint32 argCount = 2;
+    HRESULT ret;
+    void *result;
+
+    /* FIXME: dxcompiler could probably be loaded in a better spot */
+    if (dxcompiler_dll == NULL) {
+        dxcompiler_dll = SDL_LoadObject(DXCOMPILER_DLL);
+        if (dxcompiler_dll == NULL) {
+            return NULL;
+        }
+    }
+
+    if (SDL_DxcCreateInstance == NULL) {
+        SDL_DxcCreateInstance = (DxcCreateInstanceProc)SDL_LoadFunction(dxcompiler_dll, "DxcCreateInstance");
+        if (SDL_DxcCreateInstance == NULL) {
+            return NULL;
+        }
+    }
+
+    if (SDL_DxcInstance == NULL) {
+        SDL_DxcCreateInstance(&CLSID_DxcCompiler,
+                              IID_IDxcCompiler3,
+                              (void**) &SDL_DxcInstance);
+        if (SDL_DxcInstance == NULL) {
+            return NULL;
+        }
+    }
+
+    source.Ptr = hlslSource;
+    source.Size = SDL_strlen(hlslSource) + 1;
+    source.Encoding = 0; /* FIXME: The docs for this are a _bit_ scarce */
+
+    if (SDL_strcmp(shaderProfile, "ps_6_0") == 0) {
+        args[argCount++] = L"-T";
+        args[argCount++] = L"ps_6_0";
+    } else if (SDL_strcmp(shaderProfile, "vs_6_0") == 0) {
+        args[argCount++] = L"-T";
+        args[argCount++] = L"vs_6_0";
+    } else if (SDL_strcmp(shaderProfile, "cs_6_0") == 0) {
+        args[argCount++] = L"-T";
+        args[argCount++] = L"cs_6_0";
+    }
+
+    if (spirv) {
+        args[argCount++] = L"-spirv";
+    }
+
+    ret = SDL_DxcInstance->lpVtbl->Compile(
+        SDL_DxcInstance,
+        &source,
+        args,
+        argCount,
+        NULL,
+        IID_IDxcResult,
+        (void**) &dxcResult);
+
+    if (ret < 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_GPU,
+                     "IDxcShaderCompiler3::Compile failed: %X",
+                     ret);
+        return NULL;
+    } else if (dxcResult == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_GPU,
+                     "HLSL compilation failed with no IDxcResult");
+        return NULL;
+    }
```

Comment:
I believe stuff like this should call `SDL_SetError()` so that the user can retrieve the error message. Not all code in SDL seems to follow that convention, but that seems to be the exception.